File: testdata/jennies/rawtypes/variant_panelcfg_only_options/PHPRawTypes/src/VariantPanelcfgOnlyOptions/Options.php
<?php

namespace Grafana\Foundation\VariantPanelcfgOnlyOptions;

class Options implements \JsonSerializable
{
    public string $content;

    /**
     * @param string|null $content
     */
    public function __construct(?string $content = null)
    {
        $this->content = $content ?: "";
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->content !== $other->content) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{content?: string} $inputData */
        $data = $inputData;
        return new self(
            content: $data["content"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->content = $this->content;
        return $data;
    }
}

File: testdata/jennies/rawtypes/variant_panelcfg_only_options/JavaRawTypes/variant_panelcfg_only_options/Options.java
package variant_panelcfg_only_options;

import java.util.Objects;

public class Options {
    public String content;
    public Options() {
        this.content = "";
    }
    public Options(String content) {
        this.content = content;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof Options)) return false;
        Options o = (Options) other;
        if (!Objects.equals(this.content, o.content)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.content);
    }
}
